src/NLog/LogManager.cs

```diff
@@ -45,7 +45,11 @@ namespace NLog
     using NLog.Internal;
 
     /// <summary>
-    /// Creates and manages instances of <see cref="NLog.Logger" /> objects.
+    /// Creates and manages instances of <see cref="NLog.Logger"/> objects.<br/>
+    /// The purpose of LogManager is that one <see cref="LogFactory"/> can be easily used without having to carry around an instance of it.
```

Comment:
Not a big fan of very long explainations in xml-comments. Then it is better with a reference to a tutorial-page.

But I think it makes sense to explain that `LogManager wraps a singleton instance of LogFactory.`

Reply:
> Not a big fan of very long explainations in xml-comments.

I have noticed that from the very short XML documentation.

I think that the XML documentation should contain all necessary information to use a library without needing to read tutorials. The tutorial or manual should explain how a software works and how the API looks like in detail. The first part (how it works) could be described in XML comments only in a limited way (e.g. in the description of the classes). The API itself should be properly documented in XML comments, and to reduce your work the API part of the tutorial or manual could partly or fully be built from the XML comments using suitable tools.
Also, the XML documentation often is the only available documentation of a specific version of a library. The tutorials or manuals are usually adapted to reflect the latest version, and documentation of prior versions often is not kept.
Also, there may be situations when no internet is available, and unless you saved a copy of the website or the relevant pages locally, the XML documentation would be the only help you have then.
And the fact about LogManager as a LogFactory wrapper actually was not mentioned anywhere.

Reply:
I would like to give you an example of a well documented method in our software:

```
  public static partial class CMath
  {
    public static partial class Statistics
    {
      public static partial class DotGridClustering
      {
        /// ------------------------------------------------------------------
        /// <summary>
        ///   Calculate clusters of rings from the given collection of positions of objects.<br /><br />
        ///   The clusters are created by a sequence of operations:<br />
        ///   1) Find connections between the object positions using the DBSCAN method with a
        ///   search distance that is calculated from the maximum ring diameter and
        ///   the maximum object count per ring.<br />
        ///   <i> The following operations are executed twice for each cluster: </i><br />
        ///   2) Calculate the circle from the object positions of the cluster.<br />
        ///   3) Filter the object positions of the cluster based on the deviations
        ///   of [the distances of the objects from the calculated circle center]
        ///   from [the calculated circle radius].<br />
        ///   4) Calculate the circle again.<br />
        ///   5) Re-add object positions to the cluster based on the same condition as in 3).
        /// </summary>
        /// <param name="i_positions"> The positions of objects that are used to calculate the ring clusters. </param>
        /// <param name="i_ringDiameterRange"> The allowed range of the diameters of the rings. </param>
        /// <param name="i_nominalObjectCountPerRing"> The nominal number of objects on a ring. </param>
        /// <param name="i_maximumNeighborGapSize"> The maximum size of a gap between neighboring objects when searching for connections. The gap size is specified as a number of objects. </param>
        /// <param name="i_objectToCenterDistanceTolerances"> The tolerances on [the distances of the objects from the calculated circle center] against [the calculated circle radius]. The lower value must be &lt;= 0, the upper value must be &gt;= 0. </param>
        /// <param name="i_isToleranceRelative"> A flag that specifies whether the given tolerances <paramref name="i_objectToCenterDistanceTolerances" /> contain absolute (if "false") or relative (if "true") values. </param>
        /// <param name="o_clusters"> An outer array, representing the clusters, of an inner array, containing the indices of the objects positions of the cluster. </param>
        /// <returns> <see cref="CResult" /> </returns>
        /// ------------------------------------------------------------------
        public static CResult Calculate01 (IList<CLocation>   i_positions,
                                           CRange<CPhqLength> i_ringDiameterRange,
                                           int                i_nominalObjectCountPerRing,
                                           int                i_maximumNeighborGapSize,
                                           CRange<CPhqLength> i_objectToCenterDistanceTolerances,
                                           bool               i_isToleranceRelative,
                                           out int[][]        o_clusters)
        {

```
which is shown as
![grafik](https://user-images.githubusercontent.com/18487427/162606424-73d9a29e-94b4-44ca-b5be-a4e70c717483.png)

Whenever I look at this documentation, I get all relevant information that I need to use the method somewhere. But this is not the full information; the documentation of this method in our wiki is much longer. If someone needs the details, they will look there. And if someone is not interested in the details or not interested in the XML documentation... Nobody forces you to read it, everybody is free to read or just ignore it. But it should be there.

Reply:
I have adopted the school of using method-naming as comments, after seeing so many obsolete comments because methods over time have diverged in functionality. And by default when I need to understand what a method actuel does, then I read the source-code. So I prefer the xml-comment (if any) is a one-liner that explains the intention of the method with little implementation details.

Reply:
One example is #4842 where I changed the internal-logic. Where you nicely highlighted that the xml-comments was completely wrong, so they had to be tweaked #4849 + #4864